### DISCUSSION TITLE
Why don't the ply files reflect the manipulations?

### DISCUSSION THREAD

**Discussion by fabricix** [cnr-isti-vclab/meshlab#1078] [Q&A] [answered]
A have an initial slt mesh (gray in the figure 1).

I sample the mesh node points and save them in a file called `sampled-points.ply` (blue points in Figure 1).

I translate the sampled-points layer and save them in to another file `sampled-points-2.ply` (red points in Figure 2).

This looks like this:

### Figure 1
![image](https://user-images.githubusercontent.com/22794916/132248821-d5e5b0a6-6ad7-4c8d-ba84-cd6fbbc11242.png)

But when I open the  `sampled-points.ply` and  `sampled-points-2.ply` files,  they are identical:

### Figure 2
![image](https://user-images.githubusercontent.com/22794916/132249022-c67da179-bf92-4049-9afb-4fc88f3d6b72.png)

So, why the sampled-points-2 does not consider the translation in the written file? 

If I load the `sampled-point-2.ply` he does not reflect effectively the translation. Why?

How can I save the ply files in order to reflect the manipulations?

**Comment by alemuntoni** [comment, marked as answer]
The Ply format does not support transformation matrices, therefore the transformation matrix is ignored when saving the file.
You can run the "freeze current matrix" filter to apply the transformation matrix to the coordinates of your sampled mesh.

**Reply by fabricix** [reply]
Yes, thanks. If apply "Matrix: set from translation" with "freeze current matrix" option, and then save the ply file, the saved file includes the coordinate transformation. Thanks!
